src/contracts/UpgradeRegressionManager.sol

```diff
@@ -253,18 +249,20 @@ contract UpgradeRegressionManager is IUpgradeRegressionManager {
 
     _rollbackId = getRollbackId(_targets, _values, _calldatas, _description);
 
-    // Revert if the rollback is not queued for execution.
-    if (rollbackExecutableAt[_rollbackId] == 0) {
+    Rollback storage rollback = rollbacks[_rollbackId];
+    IGovernor.ProposalState _state = _getState(_rollbackId);
+
+    // Revert if the rollback has been queued
+    if (_state != IGovernor.ProposalState.Queued) {
       revert UpgradeRegressionManager__NotQueued(_rollbackId);
     }
 
     // Cancel the rollback transactions.
     for (uint256 _i = 0; _i < _targets.length; _i++) {
-      TARGET.cancelTransaction(_targets[_i], _values[_i], "", _calldatas[_i], rollbackExecutableAt[_rollbackId]);
+      TARGET.cancelTransaction(_targets[_i], _values[_i], "", _calldatas[_i], rollback.executableAt);
     }
 
-    // Remove the rollback from execution queue
-    delete rollbackExecutableAt[_rollbackId];
+    rollback.canceled = true;
```

Comment:
It *probably* shouldn't matter, but for safety and consistency, we should also effect this change before the interaction, i.e. calling the target